Add Datadog and InfluxDB integrations pages (#418)
algorithmiaio/dev-center: _pages/integrations/datadog.md

---
layout: article
title:  "Datadog"
permalink: integrations/datadog/
excerpt: "Monitor model performance metrics with Datadog"
excerpt-short: "Monitor model performance metrics with Datadog"
categories: [integrations]
tags: [integrations]
show_related: false
image:
    teaser: /language_logos/datadog.svg
---

> 

Reviewer: Lets add the enterprise only banner here like on Insights page https://algorithmia.com/developers/algorithmia-enterprise/algorithmia-insights

In this tutorial, learn how you can monitor your model performance metrics with Datadog using the new Insights feature of Algorithmia Enterprise.

> [Algorithmia Insights](https://algorithmia.com/blog/introducing-algorithmia-insights-datadog-integration) is a new feature of Algorithmia Enterprise and provides a metrics pipeline that can be used to instrument, measure, and monitor your machine learning models. Monitoring your model performance metrics can help with your organization’s overall [AI/ML governance efforts](https://algorithmia.com/blog/model-governance), and use cases include detecting [model drift](https://algorithmia.com/blog/model-drift-and-ensuring-a-healthy-machine-learning-lifecycle), data drift, and model bias.

Reviewer: ```suggestion
[Algorithmia Insights](https://algorithmia.com/blog/introducing-algorithmia-insights-datadog-integration) is a new feature of Algorithmia Enterprise that provides a metrics pipeline that can be used to instrument, measure, and monitor your machine learning models. Monitoring your model performance metrics can help with your organization’s overall [AI/ML governance efforts](https://algorithmia.com/blog/model-governance), and use cases include detecting [model drift](https://algorithmia.com/blog/model-drift-and-ensuring-a-healthy-machine-learning-lifecycle), data drift, and model bias.
```

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/insights-datadog-cc-baseline-split.png" alt="Algorithmia insights page and Datadog dashboard side-by-side" class="screenshot">

[Datadog](https://www.datadoghq.com/) is a cloud-based monitoring service that aggregates metrics and logs across different services. You can use Insights to send real-time model performance metrics from Algorithmia to Datadog as time-series data that can be included in dashboards, monitors, and alerts.

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/algorithmia-datadog.gif" alt="Switching between browser tabs with different Datadog dashboards" class="screenshot">

The [Algorithmia-Datadog integration](https://docs.datadoghq.com/integrations/algorithmia/) described in this post allows you to stream operational and inference metrics from your models in Algorithmia to Datadog, where you can create dashboards, monitors, and alerts to analyze model performance metrics in production systems across your entire model catalog.

## How the Algorithmia-Datadog integration works

> When algorithms that have Insights enabled are queried, Algorithmia emits a metrics payload to a configured Kafka broker and topic. The [Algorithmia-Datadog integration](https://docs.datadoghq.com/integrations/algorithmia/) then transforms and forwards Insights data from Kafka to the [Datadog metrics API endpoint](https://docs.datadoghq.com/api/v1/metrics/). All of the data from Algorithmia Insights will show up in Datadog as metrics with the prefix `algorithmia`. This makes it easy to include model performance data from Algorithmia Insights in dashboards or monitors.

Reviewer: ```suggestion
When algorithms that have Insights enabled are run, Algorithmia emits a metrics payload to a configured Kafka broker and topic. The [Algorithmia-Datadog integration](https://docs.datadoghq.com/integrations/algorithmia/) then transforms and forwards Insights data from Kafka to the [Datadog metrics API endpoint](https://docs.datadoghq.com/api/v1/metrics/). All of the data from Algorithmia Insights will show up in Datadog as metrics with the prefix `algorithmia`. This makes it easy to include model performance data from Algorithmia Insights in dashboards or monitors.
```

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/algorithmia-datadog-integration.png" alt="Flow chart of Datadog-Algorithmia integration for model performance metrics" class="screenshot">

The metrics payload from Algorithmia Insights contains both operational and inference metrics. Operational metrics include the algorithm name, version, owner, duration, session ID, and request ID. Inference metrics include user-defined metrics that are specified by data scientists depending on the particular algorithm and use case.

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/insights-datadog-pipeline.png" alt="Flow chart of Datadog-Algorithmia integration for operational metrics" class="screenshot">

When this integration is used, all of the operational metrics related to the algorithm name, version, owner, request ID, and session ID will be included with each data point as [tags](https://docs.datadoghq.com/getting_started/tagging/using_tags/) in Datadog. These tags can then be used to filter and group model performance metrics in dashboards and monitors.

The steps to install and configure this integration can be summarized as:

> 1. Configure Algorithmia to send Insights to a Kafka broker and topic

Reviewer: Link to https://algorithmia.com/developers/algorithmia-enterprise/algorithmia-insights in step #1

2. Install and run the [Algorithmia-Datadog integration](https://github.com/algorithmiaio/integrations/tree/master/Datadog) with your Kafka broker/topic and Datadog API key

> 3. From Algorithmia, you can then instrument, publish, and query algorithms that have Insights enabled

Reviewer: ```suggestion
3. For any published algorithm on your cluster that has Insights enabled, once the algorithm's API is called, those metrics will be pushed to DataDog. 
```

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/insights-datadog-new-timeboard.png" alt="Datadog create dashboard modal" class="screenshot">

When viewing the new dashboard, click on the ***Edit Widgets*** button. Drag a new widget such as the **Timeseries** widget onto your dashboard:

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/insights-datadog-widgets.png" alt="Datadog edit widgets view" class="screenshot">

After dragging a widget on to the dashboard, a configuration screen will appear where you can select the type of visualization you want to use and which data to show on your graph.

In the **Graph your data** configuration section, select the metric that you want to display from the list of metrics that start with the prefix `algorithmia` . For example, we’ll choose the custom `algorithmia.risk_score` metric that we defined in a model (or you could choose the default operational metric named `algorithmia.duration_milliseconds`):

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/insights-datadog-graph.png" alt="Datadog graph your data view with metric selection" class="screenshot">

> Click on the **Save** button, then the data from Insights should show up in your new widget. You might need to query your model in Algorithmia to see data flowing into Datadog.

Reviewer: ```suggestion
Click on the **Save** button, then the data from Insights should show up in your new widget. You might need to call your algorithm on your Algorithmia cluster to see data flowing into Datadog.
```

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/algorithmia-datadog.gif" alt="Switching between browser tabs with different Datadog dashboards" class="screenshot">

The [Algorithmia-Datadog integration](https://docs.datadoghq.com/integrations/algorithmia/) described in this post allows you to stream operational and inference metrics from your models in Algorithmia to Datadog, where you can create dashboards, monitors, and alerts to analyze model performance metrics in production systems across your entire model catalog.

## How the Algorithmia-Datadog integration works

When algorithms that have Insights enabled are run, Algorithmia emits a metrics payload to a configured Kafka broker and topic. The [Algorithmia-Datadog integration](https://docs.datadoghq.com/integrations/algorithmia/) then transforms and forwards Insights data from Kafka to the [Datadog metrics API endpoint](https://docs.datadoghq.com/api/v1/metrics/). All of the data from Algorithmia Insights will show up in Datadog as metrics with the prefix `algorithmia`. This makes it easy to include model performance data from Algorithmia Insights in dashboards or monitors.

> <img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/datadog/algorithmia-datadog-integration.png" alt="Flow chart of Datadog-Algorithmia integration for model performance metrics" class="screenshot">

Reviewer 2: Great job on these alt attributes. 💯
